### PR TITLE
Update docs 

### PR DESCRIPTION
**Pull request by britt6612**
This includes the latest changes to Grommet that were not added to docs.

menu.item to the docs (added in this PR https://github.com/grommet/grommet/pull/6241/files)(https://hpe.slack.com/archives/D02977518F4/p1661876566258389)
button icon kind theme stuff to the docs from this PR https://github.com/grommet/grommet/pull/6297
add `formfield.checkBox.pad` to [docs](https://github.com/grommet/grommet/pull/6300)
Add weight to Heading to [docs](https://github.com/grommet/grommet/pull/6294)

### CODE REVIEW

File: src/screens/Button.js [hunk 1]
```diff
@@ -683,6 +683,15 @@ const ButtonPage = () => (
           </PropertyValue>
         </Property>
 
+        <Property name="button.active.default.icon">
```
Comment (taysea) [inline src/screens/Button.js:686]: Do we need to add these or is this covered by the docs for `button.active.default`?

Comment (britt6612) [reply src/screens/Button.js:686]: good question I'm open to either @jcfilben ? any opinions?

Comment (jcfilben) [reply src/screens/Button.js:686]: Good point, I think we could leave `button.active.default.icon` out. Maybe on the `button.active.default` it would be good to clarify what props can be passed in there.

File: src/screens/Heading.js [hunk 2]
```diff
@@ -156,6 +156,19 @@ const HeadingPage = () => (
           </Description>
           <GenericBoolFalse />
         </Property>
+
+        <Property name="weight">
+          <Description>Font weight</Description>
+          <PropertyValue type="string">
+            <Example>"normal"</Example>
```
Comment (jcfilben) [inline src/screens/Heading.js:163]:
```suggestion
            <Example defaultValue>"normal"</Example>
```
I assume normal weight is the default value here

File: src/screens/Menu.js [hunk 3]
```diff
@@ -373,15 +359,6 @@ const MenuPage = () => (
...
 
-        <Property name="menu.group.separator.size">
-          <Description>
-            The thickness of the separator line between grouped items.
-          </Description>
-          <PropertyValue defaultValue type="string">
-            <Example>"xsmall"</Example>
-          </PropertyValue>
-        </Property>
-
```
Comment (jcfilben) [inline src/screens/Menu.js:384]: Why was this removed?

File: src/screens/FormField.js [hunk 4]
```diff
@@ -286,6 +287,11 @@ const FormFieldPage = () => (
           </PropertyValue>
         </Property>
 
+        <Property name="formField.checkBox.pad">
+          <Description>The CheckBox pad when wrapped in FormField.</Description>
+          <GenericPad />
+        </Property>
+
```
Comment (jcfilben) [inline src/screens/FormField.js:294]: Should we add this to the CheckBox doc so it is in both places?

File: src/screens/Menu.js (hunk 3 above)
Comment (britt6612) [reply src/screens/Menu.js:384]: opps mistake Ill put back

File: src/screens/FormField.js (hunk 4 above)
Comment (britt6612) [reply src/screens/FormField.js:294]: yeah we can I was back and forth on that

File: src/screens/Menu.js (hunk 3 above)
Comment (britt6612) [reply src/screens/Menu.js:384]: done I just reordered them to be alphabet

File: src/screens/Button.js [hunk 5]
```diff
@@ -679,7 +679,21 @@ const ButtonPage = () => (
             Adjustments to the default Button style when the Button is active.
           </Description>
           <PropertyValue type="object">
-            <Example>{`{}`}</Example>
+            <Example>{`{ default: {
```
Comment (taysea) [inline src/screens/Button.js:682]: We should remove the `{ default: ` outer object here

Comment (taysea) [reply src/screens/Button.js:682]: same for the rest of them

File: src/screens/Button.js [hunk 6]
```diff
@@ -903,6 +954,15 @@ const ButtonPage = () => (
           </PropertyValue>
         </Property>
 
+        <Property name="button.disabled.default.icon">
```
Comment (taysea) [inline src/screens/Button.js:957]: Can we remove this now?

File: src/screens/Button.js [hunk 7]
```diff
@@ -847,6 +889,15 @@ const ButtonPage = () => (
           <GenericExtend />
         </Property>
 
+        <Property name="button.default.icon">
```
Comment (taysea) [inline src/screens/Button.js:892]: Can we remove this now?

File: src/screens/Button.js [hunk 8]
```diff
@@ -914,6 +974,15 @@ const ButtonPage = () => (
           </PropertyValue>
         </Property>
 
+        <Property name="button.disabled.primary.icon">
```
Comment (taysea) [inline src/screens/Button.js:977]: Can we remove this now?

File: src/screens/Button.js [hunk 9]
```diff
@@ -925,6 +994,15 @@ const ButtonPage = () => (
           </PropertyValue>
         </Property>
 
+        <Property name="button.disabled.secondary.icon">
```
Comment (taysea) [inline src/screens/Button.js:997]: Remove all icon specific lines related to state since they're covered by the `button.default`, `button.secondary`, etc.

Comment (britt6612) [reply src/screens/Button.js:997]: oh yes sorry thought I pushed that up

File: src/screens/Button.js (hunk 5 above)
Comment (britt6612) [reply src/screens/Button.js]: just updated this

**Review by taysea — APPROVED** [review]
Thanks!

**Review by jcfilben — APPROVED** [review]
Looks good!
